src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const workingState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    // Clear workingState
+    if (action.type === 'clear') {
+      for (const line in workingState) {
+        delete workingState[line];
+      }
+    }
+
+    // ADD property to the workingState
+    if (action.type === 'addProperties') {
+      for (const newProperty in action.extraData) {
+        workingState[newProperty] = action.extraData[newProperty];
+      }
+    }
+
+    // Remove property from the workingState
+    if (action.type === 'removeProperties') {
+      for (const removingKey of action.keysToRemove) {
+        delete workingState[removingKey];
+      }
+    }
```

Comment:
[CODE STYLE]: Use a switch statement if you have a limited amount of conditions. It's more appropriate here than multiple if statements, as it will make your code cleaner and more readable. Also, remember to include a default case for error handling. Refactor your if conditions into a switch statement with cases for 'clear', 'addProperties', and 'removeProperties'.